core/services/job/orm.go

```diff
@@ -186,27 +186,30 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 				}
 			}
 
+			if jb.OCROracleSpec.EVMChainID == nil {
+				// If unspecified, assume we're creating a job intended to run on default chain id
+				newChain, err := o.chainSet.Default()
+				if err != nil {
+					return err
+				}
+				if newChain == nil {
+					return errors.New("internal error: chainSet.Default() should never return nil")
```

Comment:
IMHO this is still too strict of an error message because it assumes things about the system as a whole. I think we can just report what happened very plainly to the user:
```suggestion
					return errors.New("EVMChainID is nil but no default chain is available")
```

Reply:
I don't like that because it would be confusing to the user.  I think it should be as clear as possible that it's a bug that never should have been released that should be immediately reported to us, not something they did wrong.

I'm starting to think it may be better if we just don't check for nil, since we know this function can never return it anyway... it is an invariant which we should always be able to assume.  The only real purpose of it is to make sure no developer tries to change that function in a way that could return nil.  But we already have a comment about that being an invariant in the function itself, so it seems like this is not actually needed.

Reply:
If its an AssumptionViolation and should not be possible, we might consider panicking instead